python/pyarrow/_s3fs.pyx

```diff
@@ -94,6 +94,18 @@ cdef class S3FileSystem(FileSystem):
     background_writes: boolean, default True
         Whether OutputStream writes will be issued in the background, without
         blocking.
+    proxy_options: dict or str, default None
+        If a proxy is used, provide the options here. Supported options are:
+        'scheme' (str: 'http' or 'https'; required), 'host' (str; required),
+        'port' (int; required), 'username' (str; optional),
+        'password' (str; optional).
+        A proxy URI (str) can also be provided, in which case these options
+        will be derived from the provided URI.
+        The following are equivalent:
+        * S3FileSystem(proxy_options='http://username:password@localhost:8020')
+        * S3FileSystem(proxy_options={'scheme': 'http', 'host': 'localhost',
+                                      'port': 8020, 'username': 'username',
+                                      'password': 'password'})
```

Comment:
This will unfortunately not render correctly in HTML with restructuredtext/sphinx. You can either make it full text (remove the bullet points), or alternatively make it a code block, eg:

```suggestion
        The following are equivalent::
        
            S3FileSystem(proxy_options='http://username:password@localhost:8020')
            S3FileSystem(proxy_options={'scheme': 'http', 'host': 'localhost',
                                        'port': 8020, 'username': 'username',
                                        'password': 'password'})
```

Reply:
Thanks for catching done. Committed your suggestion.